tests/functional/testSubscription.js

```diff
@@ -26,6 +27,114 @@ const SUBSCRIPTION_DETAILS = {
+      created_at: new Date().toISOString()
+    };
+
+    const userDataActive = {
+      avatar: '',
+      display_name: 'Test',
+      email: 'test@mozilla.com',
+      max_devices: 5,
+      subscriptions: {vpn: {active: true}},
+      devices: [device],
+    };
+    const userDataInactive = {
+      avatar: '',
+      display_name: 'Test test',
+      email: 'test@mozilla.com',
+      max_devices: 5,
+      subscriptions: {vpn: {active: false}},
+      devices: [device]
+    };
+
+    this.ctx.guardianOverrideEndpoints = {
+      GETs: {
+        '/api/v1/vpn/account': {
+          status: 200,
+          requiredHeaders: ['Authorization'],
+          body: userDataActive
+        }
+      },
+      POSTs: {
+        '/api/v1/vpn/device': {
+          status: 201,
+          requiredHeaders: ['Authorization'],
+          bodyValidator: validators.guardianDevice,
+          callback: (req) => {
+            device.name = req.body.name;
+            device.pubkey = req.body.pubkey;
+            device.unique_id = req.body.unique_id;
+          },
+          body: {}
+        },
+      }
+    };
+
+    it('Verify subscription before enabling VPN', async () => {
+      // This test verifies the case where a user is logged in
+      // but the VPN is off when their subscription expires.
+      // When they try to turn the VPN on, they get the
+      // "Subscribe to Mozilla VPN" screen.
+
+      await vpn.authenticateInApp(true, true);
+
+      // Step 1: Override the Guardian endpoint to mock an expired subscription.
+      this.ctx.guardianOverrideEndpoints.GETs['/api/v1/vpn/account'].body =
+          userDataInactive;
+
+      // Step 2: Attempt to toggle the VPN on. This should fail.
+      await vpn.waitForQuery(queries.screenHome.CONTROLLER_TITLE.visible());
+      await vpn.clickOnQuery(queries.screenHome.CONTROLLER_TOGGLE.visible());
+
+      // Step 3: Verify that user gets the "Subscribe to Mozilla VPN" screen.
+      await vpn.waitForQuery(queries.screenHome.SUBSCRIPTION_NEEDED.visible());
+    });
+
+    it('Enable VPN with 500 error', async () => {
+      // This test verifies the case where user is logged in
+      // but the VPN is off when their subscription expires,
+      // and the client encounters an HTTP error when completing
+      // the API call to check subscription status.
+      // The expectation here is that the VPN toggles on successfully.
+
+      this.ctx.guardianOverrideEndpoints.GETs['/api/v1/vpn/account'].body =
+          userDataActive;
+
+      await vpn.authenticateInApp(true, true);
+
+      // Step 1: Override the Guardian endpoint to mock an expired subscription.
+      // Set the error status to 500.
+      this.ctx.guardianOverrideEndpoints.GETs['/api/v1/vpn/account'].body =
+          userDataInactive;
+      this.ctx.guardianOverrideEndpoints.GETs['/api/v1/vpn/account'].status =
+          500;
+
+      await vpn.activate();
+      await vpn.waitForCondition(async () => {
```

Comment:
I would prefer if this waitForCondition was for the `Connecting ...` text not the `VPN is on.` It's not expected that the VPN would successfully connect in this state.

Reply:
From my conversation with @bakulf this morning, I believe the VPN should successfully connect in this case. Baku, can you confirm?